dotCMS/src/integration-test/java/com/dotmarketing/util/UtilHTMLTest.java

```diff
@@ -0,0 +1,232 @@
+        when(contentlet.getContentType()).thenReturn(type);
+        when(type.baseType()).thenReturn(any);
+
+        final String iconClass = UtilHTML.getIconClass(contentlet);
+        assertNotNull(iconClass);
+        assertEquals("pageIcon", iconClass);
+    }
+
+    /**
+     * Method to test: getIconClass
+     * Given Scenario: Given a key value base type
+     * ExpectedResult: keyValueIcon icon should be returned
+     */
+    @Test
+    public void test_getIconClass_keyvalue () throws DotDataException {
+
+        final Contentlet contentlet = mock(Contentlet.class);
+        final ContentType type      = mock(ContentType.class);
+        final BaseContentType any   = BaseContentType.KEY_VALUE;
+        when(contentlet.getContentType()).thenReturn(type);
+        when(type.baseType()).thenReturn(any);
+
+        final String iconClass = UtilHTML.getIconClass(contentlet);
+        assertNotNull(iconClass);
+        assertEquals("keyValueIcon", iconClass);
+    }
+
+    /**
+     * Method to test: getIconClass
+     * Given Scenario: Given a persona base type
+     * ExpectedResult: personaIcon icon should be returned
+     */
+    @Test
+    public void test_getIconClass_persona () throws DotDataException {
+
+        final Contentlet contentlet = mock(Contentlet.class);
+        final ContentType type      = mock(ContentType.class);
+        final BaseContentType any   = BaseContentType.PERSONA;
+        when(contentlet.getContentType()).thenReturn(type);
+        when(type.baseType()).thenReturn(any);
+
+        final String iconClass = UtilHTML.getIconClass(contentlet);
+        assertNotNull(iconClass);
+        assertEquals("personaIcon", iconClass);
+    }
+
+    /**
+     * Method to test: getIconClass
+     * Given Scenario: Given a vanity url base type
+     * ExpectedResult: vanityIcon icon should be returned
+     */
+    @Test
+    public void test_getIconClass_vanity () throws DotDataException {
+
+        final Contentlet contentlet = mock(Contentlet.class);
+        final ContentType type      = mock(ContentType.class);
+        final BaseContentType any   = BaseContentType.VANITY_URL;
+        when(contentlet.getContentType()).thenReturn(type);
+        when(type.baseType()).thenReturn(any);
+
+        final String iconClass = UtilHTML.getIconClass(contentlet);
+        assertNotNull(iconClass);
+        assertEquals("vanityIcon", iconClass);
+    }
+
+    /**
+     * Method to test: getIconClass
+     * Given Scenario: Given a dotasset base type and txt file
+     * ExpectedResult: txtIcon icon should be returned
+     */
+    @Test
+    public void test_getIconClass_dotasset_txt () throws DotDataException, IOException {
+
+        final Contentlet contentlet = mock(Contentlet.class);
+        final ContentType type      = mock(ContentType.class);
+        final BaseContentType any   = BaseContentType.DOTASSET;
+        File txtFile                = FileUtil.createTemporalFile("testtxt", ".txt");
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'txtFile' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5436906)